### PR TITLE
Fetches only the Lois and Submissions that the current user has access to

### LINKED ISSUE #2058: [Export CSV] Data collectors can export data collected by others

### PR DESCRIPTION
**Pull request by rfontanarosa**
closes #2058

### CODE REVIEW

File: functions/src/common/auth.ts [hunk 1]
```diff
@@ -107,3 +107,10 @@ export function canImport(
   const role = getRole(user, survey);
   return !!role && [Pb.Role.SURVEY_ORGANIZER].includes(role);
 }
+
+export function canManage(
```
Comment (amysorto) [inline functions/src/common/auth.ts:111]: It seems like this just calls `canImport`, maybe it makes sense to just call it directly instead?

File: functions/src/common/datastore.ts [hunk 2]
```diff
@@ -132,25 +132,57 @@ export class Datastore {
...
+    jobId: string,
+    userId?: string
+  ): Promise<DocumentData[]> {
+    if (!userId) {
+      return (
+        await this.db_
+          .collection(lois(surveyId))
+          .where(l.jobId, '==', jobId)
+          .get()
+      ).docs;
```
Comment (amysorto) [inline functions/src/common/datastore.ts:169]: Is this function being called elsewhere in the application currently? I am assuming `.docs` being added at the end here might cause some issues.

**Review by amysorto — DISMISSED** [review]

File: functions/src/common/datastore.ts (hunk 2 above)
Comment (rfontanarosa) [reply functions/src/common/datastore.ts:169]: nope, it is called only there. something similar have been done by data-store.service.ts under web, but there I could use combineLatest to merge results. I didn't find a better way. what do you think @gino-m?

```
  getAccessibleLois$(
    {id: surveyId}: Survey,
    userId: string,
    canManageSurvey: boolean
  ): Observable<List<LocationOfInterest>> {
    if (canManageSurvey) {
      return this.db
        .collection(`${SURVEYS_COLLECTION_NAME}/${surveyId}/lois`)
        .valueChanges({idField: 'id'})
        .pipe(map(this.toLocationsOfInterest));
    }

    const importedLois = this.db.collection(
      `${SURVEYS_COLLECTION_NAME}/${surveyId}/lois`,
      ref => ref.where(l.source, '==', Source.IMPORTED)
    );

    const fieldDataLois = this.db.collection(
      `${SURVEYS_COLLECTION_NAME}/${surveyId}/lois`,
      ref =>
        ref
          .where(l.source, '==', Source.FIELD_DATA)
          .where(l.ownerId, '==', userId)
    );

    return combineLatest([
      importedLois.valueChanges({idField: 'id'}),
      fieldDataLois.valueChanges({idField: 'id'}),
    ]).pipe(
      map(([predefinedLois, fieldDataLois]) =>
        this.toLocationsOfInterest(predefinedLois.concat(fieldDataLois))
      )
    );
  }
```

Comment (gino-m) [reply functions/src/common/datastore.ts:169]: Lgtm

Comment (gino-m) [reply functions/src/common/datastore.ts:169]: Is the idea here that you're trying run both queries in parallel? Is the additional complexity of all the stream logic worth it? Perhaps this is an over-optimization?

**Review by gino-m — APPROVED** [review]

**Pull request merged**